src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.Com.cs

```diff
@@ -207,21 +207,78 @@ public static unsafe void GetNativeVariantForObject(object? obj, IntPtr pDstNati
                 case System.Reflection.Missing value:
```

Comment:
You can change the `VariantWrapper value` can above to throw ArgumentException too. (It throws ArgumentException on CoreCLR.)